src/plugins/discover/public/application/components/default_discover_table/helper.tsx

```diff
@@ -94,7 +94,9 @@ export function getLegacyDisplayedColumns(
       colRightIdx: idx + 1 >= columns.length ? -1 : idx + 1,
     };
   });
-  return !hideTimeField && indexPattern.timeFieldName
+  return !hideTimeField &&
+    indexPattern.timeFieldName &&
+    !columns.includes(indexPattern.timeFieldName)
```

Comment:
nit: consider assigning this to a meaningfully named variable so the reader doesn't have to parse out what this is trying to do

Reply:
Will think a good name and update